charms/knative-serving/tests/unit/test_charm.py

```diff
@@ -186,6 +190,35 @@ def test_context_changes(harness):
         assert harness.charm._context == context
 
 
+@pytest.mark.parametrize(
+    "custom_image_config, expected_custom_images",
+    [
+        (
+            yaml.dump({"name1": "image1", "name2": "image2"}),
+            {"name1": "image1", "name2": "image2"},
+        ),
+        (
+            yaml.dump({}),
+            {},
+        ),
+    ],
+)
+def test_custom_images_config_context(custom_image_config, expected_custom_images, harness):
+    """Asserts that the custom_images context is as expected.
+
+    Note: This test is trivial now, where custom_image_config always equals custom_images.  But
```

Comment:
```suggestion
    Note: This test is trivial now, where custom_image_config always equals custom_images,
    but ...
```

Reply:
I'm not entirely sure what you mean with this note, does this mean this implementation will change in the near future?

Reply:
yeah I think when we have rocks for this, this dict will be where they go.  For now we have no custom images, so we just use upstream's

Reply:
Tried rewording a little, but not sure its better